lib/stream.js

```diff
@@ -92,8 +93,9 @@ const getStreamPromise = (stream, {encoding, buffer, maxBuffer}) => {
 		return getStream(stream, {maxBuffer});
 	}
 
-	if (encoding === null || encoding === 'buffer') {
-		return getStreamAsBuffer(stream, {maxBuffer});
+	if (encoding === 'buffer') {
+		const streamBufferPromise = getStreamAsBuffer(stream, {maxBuffer});
+		return streamBufferPromise.then(value => bufferToUint8Array(value));
```

Comment:
We can use [`getStreamAsArrayBuffer()`](https://github.com/sindresorhus/get-stream#getstreamasarraybufferstream-options) here to completely remove the dependency on `Buffer`. It still requires converting to an `Uint8Array` though.